### PR TITLE
[Xedra Evolved] Electronic home doors

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
None

#### Purpose of change
Adds some electronically locked doors to Xedra Evolved homes that can be opened if the player guesses the code, which is handled by a contested roll.

#### Describe the solution
Adds doors and ter transform and eocs.  Adds the new doors to existing palettes.

#### Describe alternatives you've considered
Adding the new doors as variant doors for vanilla and then overwriting them to have an eoc option in Xedra Evolved.

#### Testing
Success!

#### Additional context
failure
success

### CODE REVIEW

**Comment by Maleclypse** [comment]
@Ramza13 Are mod added skills not possible to use with the contested rolls?  
```
{
          "id": "eoc_unlock_elock",
          "condition": { "roll_contested":
                                          ^
                                           { "u_val": "deduction" }, "difficulty": 12 },
          "effect": [
            { "u_location_variable": { "global_val": "door_transform" }, "min_radius": 1, "max_radius": 2 },
```

error says that `deduction` isn't acceptable if I'm reading it correctly.

File: data/mods/Xedra_Evolved/furniture_and_terrain/terrain-doors.json [hunk 1]
```diff
@@ -40,5 +40,211 @@
...
+      "LOCKED",
+      "BLOCK_WIND",
+      "SUPPORTS_ROOF"
+    ],
+    "examine_action": {
+      "type": "effect_on_condition",
+      "effect_on_conditions": [
+        {
+          "id": "eoc_unlock_elock",
+          "condition": { "roll_contested": { "u_val": "deduction" }, "difficulty": 12 },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/furniture_and_terrain/terrain-doors.json:71]:
```suggestion
          "condition": { "roll_contested": { "u_val": "skill_level", "skill": "deduction" }, "difficulty": 12 },
```

File: data/mods/Xedra_Evolved/furniture_and_terrain/terrain-doors.json [hunk 2]
```diff
@@ -40,5 +40,211 @@
...
+      "BLOCK_WIND",
+      "SUPPORTS_ROOF"
+    ],
+    "open": "t_door_o_peep",
+    "examine_action": {
+      "type": "effect_on_condition",
+      "effect_on_conditions": [
+        {
+          "id": "eoc_unlock_elock",
+          "condition": { "roll_contested": { "u_val": "deduction" }, "difficulty": 12 },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/furniture_and_terrain/terrain-doors.json:141]:
```suggestion
          "condition": { "roll_contested": { "u_val": "skill_level", "skill": "deduction" }, "difficulty": 12 },
```

**Comment by Ramza13** [comment]
> @Ramza13 Are mod added skills not possible to use with the contested rolls?
> 
> ```
> {
>           "id": "eoc_unlock_elock",
>           "condition": { "roll_contested":
>                                           ^
>                                            { "u_val": "deduction" }, "difficulty": 12 },
>           "effect": [
>             { "u_location_variable": { "global_val": "door_transform" }, "min_radius": 1, "max_radius": 2 },
> ```
> 
> error says that `deduction` isn't acceptable if I'm reading it correctly.

I'm on vacation till Tuesday but I'll check it out when I get back.

**Comment by Maleclypse** [comment]
> > @Ramza13 Are mod added skills not possible to use with the contested rolls?
> > ```
> > {
> >           "id": "eoc_unlock_elock",
> >           "condition": { "roll_contested":
> >                                           ^
> >                                            { "u_val": "deduction" }, "difficulty": 12 },
> >           "effect": [
> >             { "u_location_variable": { "global_val": "door_transform" }, "min_radius": 1, "max_radius": 2 },
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > error says that `deduction` isn't acceptable if I'm reading it correctly.
> 
> I'm on vacation till Tuesday but I'll check it out when I get back.

No stress, I've figured out what I was doing wrong and now everything is working excellent.  Thank you for adding this, I'm going to be coming up with lots of uses for it.  <3

**Pull request merged**